source/rust_verify/example/guide/nonlinear_bitvec.rs

```diff
@@ -0,0 +1,66 @@
+#[allow(unused_imports)]
+use builtin_macros::*;
+#[allow(unused_imports)]
+use builtin::*;
+
+verus! {
+
+// ANCHOR: bound_checking
```

Comment:
This example is confusing.  What is the `mul` function for?  Why do we want to prove this?  What are the assertions on lines 22 and 23 doing?  Are they necessary for the proof or are they making some other point?  

Perhaps we should use an inline version of a lemma like [LemmaMulUpperBound](https://github.com/verus-lang/verus/blob/2dc6a17858e241ded90617375d9ce2784576afb7/source/rust_verify/example/integer_ring/integer_ring_bound_check.rs#L24)

Also, instead of saying
```
// note that y <= 0xffff is not available inside this proof block
```
perhaps:
```
// assert(y <= 0xffff);    <- Fails, since this property is not included in the `requires` clause
```